### PR TITLE
Add optional languageId to window.createOutputChannel API (#19561)

### LINKED ISSUE #19561: Allow to provide colorizers for specific output channels
- VSCode Version: 1.8
- OS Version: All

One user of my [Code Runner](https://marketplace.visualstudio.com/items?itemName=formulahendry.code-runner) extension reported this issue: https://github.com/formulahendry/vscode-code-runner/issues/59

**Steps to Reproduce**:

1. Install [Output Colorizer ](https://marketplace.visualstudio.com/items?itemName=IBM.output-colorizer)
2. Install [Code Runner](https://marketplace.visualstudio.com/items?itemName=formulahendry.code-runner)

**Result**: 

IBM's Output Colorizer extension is no longer working.

Both of us using `text/x-code-output` (https://github.com/formulahendry/vscode-code-runner/blob/master/package.json#L197 , https://github.com/IBM-Bluemix/vscode-log-output-colorizer/blob/master/package.json#L4)
Not sure there is a way to avoid the color highlight of one extension overwriting another? Could the color highlight exsit for both extension?

### PR DESCRIPTION
**Pull request by gjsjohnmurray**
This PR fixes #19561 by implementing proposed API which adds an optional `languageId` param to `window.createOutputChannel`.

This was previously [suggested](https://github.com/microsoft/vscode/issues/11005#issuecomment-243043560) by @ArtemGovorov and was [supported](https://github.com/microsoft/vscode/issues/11005#issuecomment-243055560) at the time by @jrieken. 

It was later raised as a solution for #19561, opened by @formulahendry.

### CODE REVIEW

File: src/vs/workbench/api/common/extHostOutput.ts [hunk 1]
```diff
@@ -117,24 +117,24 @@ export class ExtHostOutputService implements ExtHostOutputServiceShape {
 		}
 	}
 
-	createOutputChannel(name: string, extension: IExtensionDescription): vscode.OutputChannel {
+	createOutputChannel(name: string, languageId: string, extension: IExtensionDescription): vscode.OutputChannel {
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHostOutput.ts:120]: proposed API must be enforced

File: src/vs/workbench/api/common/extHost.api.impl.ts [hunk 2]
```diff
@@ -648,8 +648,11 @@ export function createApiFactoryAndRegisterActors(accessor: ServicesAccessor): I
 			withProgress<R>(options: vscode.ProgressOptions, task: (progress: vscode.Progress<{ message?: string; worked?: number }>, token: vscode.CancellationToken) => Thenable<R>) {
 				return extHostProgress.withProgress(extension, options, task);
 			},
-			createOutputChannel(name: string): vscode.OutputChannel {
-				return extHostOutputService.createOutputChannel(name, extension);
+			createOutputChannel(name: string, languageId?: string): vscode.OutputChannel {
+				if (languageId) {
+					checkProposedApiEnabled(extension, 'outputChannelLanguage');
```
Comment (gjsjohnmurray) [inline src/vs/workbench/api/common/extHost.api.impl.ts:653]: @jrieken is this enforcement insufficient or incorrect?

Comment (jrieken) [reply src/vs/workbench/api/common/extHost.api.impl.ts:653]: Yeah, that will do it's job

**Comment by gjsjohnmurray** [comment]
@sandy081 any chance you could consider merging this in time for Monday's endgame?

**Comment by sandy081** [comment]
I am sorry that I have to postpone this for the next milestone as this has to be discussed in our API sync meeting first. The last API sync for this milestone finished today and unfortunately I missed checking this in time. Since there is no urgency, I will plan this for next milestone and discuss in our API sync.

**Comment by gjsjohnmurray** [comment]
@sandy081 no worries. I know you have been busy with other things.

**Comment by gjsjohnmurray** [comment]
@sandy081 now that 1.64 has shipped, when will the next API sync meeting happen? I'd like to get this into Insiders soon.

**Comment by sandy081** [comment]
@gjsjohnmurray Thanks for reminding. API call is today. As per the process we will first add it to the proposed API and after getting some feedback we will finalize it. So most probably this will be a proposed API in this month and goes into finalization next month.

**Review by sandy081 — APPROVED** [review]
Changes look good. Did some small clean ups. Approved.

**Pull request merged**